### ISSUE TITLE
Performance/Sample cop incorrectly reports and auto-corrects var.shuffle.join

### ISSUE THREAD

**Issue by DMA57361** [bbatsov/rubocop#1816]
After updating to 0.30.1 we've had the following new offence come up, which I believe is incorrect and causes a change in behaviour when auto-correcting:

`Performance/Sample: Use sample instead of shuffle.join`

When auto-correcting, this changes `variable.shuffle.join` into `variable.sample`.
This behaviour is very different:

```
2.2.0 :001 > ('A'..'Z').to_a.shuffle.join
 => "HKUFLXGYBTNVQMRZCJEIPSOWDA"
2.2.0 :002 > ('A'..'Z').to_a.sample
 => "M"
```

**Comment by bbatsov** [comment]
Definitely a bug.

**Comment by bbatsov** [comment]
//cc @rrosenblum

**Comment by rrosenblum** [comment]
Yes, this is a bug. I apologize, I was missing adequate negative test cases during the last change. I think that I can get a fix together quickly.

Thank you for submitting the issue.
